app/transport/http/server.go

```diff
@@ -2,26 +2,31 @@ package http
 
 import (
 	"fmt"
+	"github.com/go-devs-ua/octagon/cfg"
+	"github.com/gorilla/mux"
 	"net/http"
+	"time"
 )
 
+const timeoutMsg = "Connection timeout"
+
 // Server is simple server
 type Server struct{ *http.Server }
 
 // NewServer will initialize the server
-// that would be router type agnostic
-// we can switch to any router type
-// that implements Router interface
-func NewServer(uu UserUsecase, r Router) *Server {
-	hdl := NewUserHandler(uu)
-	r.mapRoutes(hdl)
-	// TODO: Add config
-	srv := &http.Server{
-		Addr:    ":8080",
-		Handler: r,
-	}
+func NewServer(opt cfg.Options, handler UserHandler) *Server {
+	router := new(mux.Router)
+
+	attachUserEndpoints(router, handler)
 
-	return &Server{srv}
+	return &Server{
+		Server: &http.Server{
+			Addr:         opt.Server.Host,
```

Comment:
Here I see that you do not use the value of port. And the field Addr will look like "localhost". But it seems that it should looks like "localhost:8080 and probably the server even do will not start